db/models.py

```diff
@@ -45,3 +59,68 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(to=User, on_delete=models.CASCADE)
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        to=MovieSession,
+        on_delete=models.CASCADE
+    )
+    order = models.ForeignKey(to=Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["row", "seat"],
+                name="row_seat"
+            )
+        ]
+
+    def __str__(self) -> str:
+        return f"{self.movie_session.movie.title} " \
+               f"{self.movie_session.show_time} (row: {self.row}, " \
+               f"seat: {self.seat})"
+
+    def clean(self) -> None:
+        print(self.movie_session.cinema_hall.rows)
+        if not (1 <= self.row <= self.movie_session.cinema_hall.rows):
+            raise ValidationError({
+                "row": f"row number must be in available range: (1, rows):"
+                       f" (1, {self.movie_session.cinema_hall.rows})"
+            })
+
+        if not (
+                1 <= self.seat <= self.movie_session.cinema_hall.seats_in_row
+        ):
+            raise ValidationError({
+                "seat": f"seat number must be in available range: "
+                        f"(1, seats_in_row):"
+                        f" (1, {self.movie_session.cinema_hall.seats_in_row})"
+            })
+
+    def save(
+            self, force_insert: bool = False,
+            force_update: bool = False,
+            using: Any = None,
+            update_fields: Any = None) -> Any:
```

Comment:
If there's a newline after the opening parenthesis, there should also be a newline before the closing one.